### PR TITLE
Hyperparameter Tuning

### PR DESCRIPTION
**Pull request by DhrubaPatra**
Changed the hyperparameters to bring the validation loss to 2.00.

### CODE REVIEW

**Comment by gantrol** [comment]
It seems you use something like `optuna`. I think you should not change `context_length` for better result.

**Comment by karpathy** [comment]
Thank you! I adopted your changes and can get better performance:

- LR 1e-3 -> 7e-4
- BS 64 -> 128
- emb size -> 48

I think I'd like to not change the `n` in the n-gram to be larger than what we explore in the `ngram` repo. So I'll keep it at 3 so it's more directly comparable. It's very interesting that it can be pushed to higher numbers and with much better results!

Pushed commit, closing this.

**Comment by DhrubaPatra** [comment]
Hey can i be added as a contributer @karpathy 😓
